lib/bolt/transport/ssh/connection.rb

```diff
@@ -275,6 +312,32 @@ def validate_ssh_version
             raise "Powershell over SSH requires OpenSSH server >= 7.9, target is running #{match[1]}"
           end
         end
+
+        def handle_net_ssh_error(err)
+          match = /could not settle on (?<algo>encryption|hmac|host_key|kex)/.match(err.message)
+
+          if match
+            enabled = Net::SSH::Transport::Algorithms::DEFAULT_ALGORITHMS[match[:algo].to_sym]
+            disabled = Net::SSH::Transport::Algorithms::ALGORITHMS[match[:algo].to_sym] - enabled
+
+            message = "#{err.message}\n"\
+                      "Supported algorithms are defined by the Ruby net-ssh library. All supported, "\
+                      "non-deprecated algorithms are enabled by default. Disabled algorithms can be "\
+                      "enabled by setting the '#{match[:algo].gsub('_', '-')}-algorithms' config option.\n"\
+                      "Enabled algorithms: #{enabled.join(',')}\n"\
```

Comment:
In practice, it's hard to follow how this information relates to the error when it occurs. Maybe it makes sense in the case where you *haven't* specified a list of preferred algorithms, but when you did, it's weird that it shows the list that you *could* have selected.

For context, with `encryption-algorithms: [ "blowfish-ctr" ]`
```
  could not settle on encryption_client algorithm
  Server encryption_client preferences: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc,blowfish-cbc,cast128-cbc,3des-cbc
  Client encryption_client preferences: blowfish-ctr
  Supported algorithms are defined by the Ruby net-ssh library. All supported, non-deprecated algorithms are enabled by default. Disabled algorithms can be enabled by setting the 'encryption-algorithms' config option.
  Enabled algorithms: aes256-ctr,aes192-ctr,aes128-ctr
  Disabled algorithms: aes256-cbc,aes192-cbc,aes128-cbc,rijndael-cbc@lysator.liu.se,blowfish-ctr,blowfish-cbc,cast128-ctr,cast128-cbc,3des-ctr,3des-cbc,idea-cbc,none
```